src/Core/Entities/PlayerEntity.m

```diff
@@ -7913,8 +7913,21 @@ - (NSString *) primedEquipmentName:(NSInteger)offset
 
 - (BOOL) setPrimedEquipment:(NSString *)eqKey
 {
+	NSUInteger c = [eqScripts count];
+	NSUInteger current = primedEquipment;
 	primedEquipment = [self eqScriptIndexForKey:eqKey];	// if key not found primedEquipment is set to primed-none
-	return (primedEquipment != [eqScripts count]);
+	BOOL result = YES;
+	if (primedEquipment == c)
```

Comment:
If we changed this to `if (primedEquipment == c && ![eqKey isEqualToString:@""])`, we could make the setPrimedEquipment method take care of un-priming everything too. See below proposed change to the message display line.